zephyr/lib/cpu.c

```diff
@@ -149,19 +123,13 @@ int cpu_enable_core(int id)
 	 * and the idle thread stack.
 	 */
 	if (pm_state_next_get(id)->state == PM_STATE_ACTIVE)
-		z_init_cpu(id);
+		cpu_resume = false;
```

Comment:
could just do
```
	if (pm_state_next_get(id)->state == PM_STATE_ACTIVE) {
		k_smp_cpu_start(id, secondary_init, NULL);
		return 0;
	}
```
here and avoid adding `cpu_resume` altogether.

Reply:
need to consider #if ZEPHYR_VERSION(3, 0, 99) <= ZEPHYR_VERSION_CODE

Reply:
thanks, updated